### PR TITLE
Fix character picker bugs

### PR DESCRIPTION
**Pull request by djgrant**
Bugs fixed:
- When choosing no after being offered new letters, the character picker for the reverted letters does not show any charcter being in use
- Being possible to change to an in use character

To check bugs are fixed:
- Add the book 'DEE' to your basket twice
- Choose no
- Click on a character and click again to open the picker
- Observe that the correct character is in use
- Try to change the character to the other in use character (shouldn't work)

### CODE REVIEW

File: src/js/steps/letters/generateOverlay.js [hunk 1]
```diff
@@ -167,21 +167,18 @@ module.exports = function (options, $events) {
...
+        data.combinedLetters
+          .filter(letter => letter.changed === true)
+          .reduce((arr, letter) => {
+            return arr.concat(
+              letter.characters.filter(character => character.character === letter.default_character)
+            );
+          }, [])
+          .forEach((letter, i) => {
+            var $letter = $(`.letter[data-letter=${letter.letter}][data-character=${letter.character}]`);
+            data.changeCharacter(++i, letter, $letter, false);
```
Comment (callumacrae) [inline src/js/steps/letters/generateOverlay.js]: Is there any difference between using this and just `i + 1`? It seems to be the only usage in this scope

**Comment by callumacrae** [comment]
lgtm :+1:

**Comment by callumacrae** [comment]
Dude, please don't merge PRs that break the build!

**Comment by djgrant** [comment]
I'm pretty sure the build didn't fail over that change!

**Pull request merged**